subsys: net: zperf: fix the calculation ratio between mbps, kbps and bps

fengming-ye opened the pull request:
The ratio between mbps and kbps, kbps and bps should be 1000, instead of 1024, as common sense. The wrong ratio will decrease the Zperf throughput result.

fengming-ye committed:
net: zperf: fix the calculation ratio between mbps, kbps and bps

The ratio between mbps and kbps, kbps and bps should be 1000, instead of
1024, as common sense.
The wrong ratio will decrease the Zperf throughput result.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

jukkar commented on subsys/net/lib/zperf/zperf_shell.c:
```diff
@@ -76,9 +76,9 @@ static struct in_addr shell_ipv4;
 
```
Looks ok, please change the commit subject to `net: zperf: ....` as the net is the subsystem so there is no need to have subsys prefix there.

jukkar replied:
Another option (if we want to keep using 1024), is to change the printouts to `Kibit/s` or `Mibit/s` etc, this way the values would not change and could be compared with the earlier results.

fengming-ye replied:
Hi jukkar,

It will be confusing when common use is Kbps(like iperf) but zperf uses Kibits/s.

jukkar replied:
> And it will be confusing when common use is Kbps(like iperf) but zperf uses Kibits/s.

Yes, I agree, it would be nice if we can compare the iperf numbers.

I think we could add a note to migration guide ` doc/releases/migration-guide-3.6.rst` about this so that it is clear that the performance values change.  Could you add such a commit?

fengming-ye replied:
Added commit is this PR

rlubos commented:
@fengming-ye Please split the line in the commit message, it's too long:
>  3: UC4 Commit message body line exceeds max length (95>75): "The ratio between mbps and kbps, kbps and bps should be 1000, instead of 1024, as common sense."

fengming-ye commented:
> @fengming-ye Please split the line in the commit message, it's too long:
> 
> > 3: UC4 Commit message body line exceeds max length (95>75): "The ratio between mbps and kbps, kbps and bps should be 1000, instead of 1024, as common sense."

Fixed

dleach02 reviewed (dismissed):
@fengming-ye, lets follow @jukkar request to update the migration guide

fengming-ye committed:
doc: migration-guide: 3.6: zperf statistics changes

Add information about zperf ratio between mbps and kbps, kbps and bps
changes from 1024 to 1000, to align with iperf.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

henrikbrixandersen commented:
@fengming-ye Please rebase on `main` and resolve any merge conflicts.

fengming-ye commented:
> @fengming-ye Please rebase on `main` and resolve any merge conflicts.

rebased
